source/_components/tts.markdown

```diff
@@ -23,28 +23,66 @@ tts:
-|---------------------|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `cache` | True    | Allow TTS to cache voice file to local storage. |
-| `cache_dir`  | tts      | Folder name or path to a folder for caching files. |
-| `time_memory`     | 300     | Time to hold the voice data inside memory for fast play on a media player. Minimum is 60 s and the maximum 57600 s (16 hours). |
+{% configuration %}
+cache:
+  description: Allow TTS to cache voice file to local storage.
+  required: false
+  type: boolean
+  default: True
+cache_dir:
+  description: Folder name or path to a folder for caching files.
+  required: false
+  type: string
+  default: tts
+time_memory:
+  description: Time to hold the voice data inside memory for fast play on a media player. Minimum is 60 s and the maximum 57600 s (16 hours).
+  required: false
+  type: int
+  default: 300
+base_url:
+  description: A base URL to use *instead* of the one set in the [http component](/components/http/). It is used as-is by the `tts` component. In particular, you need to include the protocol scheme `http://` or `https://` and the correct port number. They will not be automatically added for you.
+  required: false
+  type: string
+  default: value of ``http.base_url``
+{% endconfiguration %}
 
 The extended example from above would look like the following sample:
 
 ```yaml
-# Example configuration.yaml entry for google tts service
+# Example configuration.yaml entry for Google TTS service
 tts:
   - platform: google
     cache: true
     cache_dir: /tmp/tts
     time_memory: 300
+    base_url: http://192.168.0.10:8123
 ```
 
+## {% linkable_title When do you need to set `base_url` here? %}
+
+The general answer is "whenever the global `base_url` set in [http component](/components/http/) is not adequate to allow the `say` service to run". The `say` service operates by generating a media file that contains the speech corresponding to the text passed to the service. Then the `say` service sends a message to the media device with a URL pointing to the file. The device fetches the media file at the URL and plays the media. Some combinations of a media device, network configuration and Home Assistant configuration can make it so that the device cannot fetch the media file.
+
+The following sections describe some of the problems encountered with media devices.
+
+### {% linkable_title Self-signed certificates %}
```

Comment:
The section is not advocating the use of self-signed certificates. It is telling users that if they do use a self-signed certificate, that may cause problems when they try to use the `say` service with their media device. If anything, I'd expect it would make some folks switch from using self-signed certificates in favor of using Let's Encrypt.

Reply:
I do agree with you, @pvizeli, nevertheless, this piece of documentation help users who DO.

`This problem occurs when your Home Assistant instance is configured to be accessed through SSL, and you are using a self-signed certificate.`

There is no recommendation in this.